tests/integration/utils/ai-gateway-helpers.ts

```diff
@@ -0,0 +1,110 @@
+import { TestContext } from 'vitest'
```

Comment:
Just pointing out that the pattern of having a util file for a specific chunk of tests, away from the test file itself, is not a pattern I've seen in this repo. Same goes for the usage of a function (`createAIGatewayCheckFunction`) to generate the functions to be tested, which feels like yet another abstraction on top of the site builder that may make tests a bit harder to follow.

Not an issue necessarily, and definitely not a blocker.